### ISSUE TITLE
Option for disabling threading feature

### ISSUE THREAD

**Issue by gecko655** [integrations/slack#1500]
**Describe the solution you'd like**
Recently (within last 2 hours?), Issue comments are posted as thread posts of their issue creation post.
This feature is described here: https://github.com/integrations/slack#threading

This behavior is too annoying because I missed important comments that are commented to issues and are important for all staffs.
I want to disable any threading feature, so that all issue comments are posted in main slack timeline.

**Describe alternatives you've considered**
Just reverting threading feature also solves this problem.

**Additional context**
Screenshot of current threading feature, rolled out in last 2 hours.

<img width="539" alt="スクリーンショット 2022-10-18 16 26 37" src="https://user-images.githubusercontent.com/6166778/196364582-0cee9d00-0f40-4a63-969c-61d54d724a66.png">

**Comment by ashokirla** [comment]
Hi @gecko655 👋 
Yes, we have recently enabled threading feature for issues. The same will be applied to PRs as well. The main reason for moving the notifications of issue into thread is to reduce the noise in the channel. We heard from multiple customers that the channel has become pretty much unusable for other communications. After careful evaluation we have upgraded our notifications to thread so that all the context is in one place for users to track.

Regarding your concern, you will not be missing any comments that happen on Slack. Any participant of the thread will be automatically notified in slack. This is the native behaviour of slack and most of the apps take advantage of this functionality to ensure only relevant folks are notified. We have another feature called 'mentions' that ensures any person mentioned in the issue parent card or subsequent card is already considered as participant in the thread. And you will get notified.

**Comment by Ovis** [comment]
With this feature enabled, the person who created the Issue will also be notified when the comment is posted.
However, for my purposes, just having it posted in Slack is sufficient.
Also, I think it is unnecessary for Slack to send a notification to the person who made the post on GitHub.
It would be very inconvenient for me to be notified by replies unnecessarily.

**Comment by tamakiii** [comment]
I don't think it's noise, but valuable information. This change of behavior is making me and our team being actually confused. Information design shoud not be forced by tool.

**Comment by gecko655** [comment]
Actually my colleagues and I are failed to follow the conversions, which I am not assigned, mentioned or I have not commented to.
I need such "noise"s to get "what's going on in my whole team".

These "noise"s can:
- prevent some Issues/Pull Requests from being approved that should not be approved, but assignee does not notice and other irrelevant staff notices.
- let some staffs join the conversion that is not assigned/mentioned to them but is actually relevant to them.

Could you reconsider the situation and implement the option, please?

**Comment by ashokirla** [comment]
Thanks everyone for your feedback. Threading is the just one part of the bigger story where we want to bring in collaborative aspect by providing ability to act upon the issues and also participate in the discussion in the place where you have all the context and history. 

Moreover, as part of issue notifications, create issue, close issue and reopen issue notifications are also shared to the main channel. Only the comments are limited to the thread and not broadcasted. We are being opinionated here and only want to get the attention of participants, mentioned and assignees. 

If others who are not involved and want to track comments in a specific issue (which is not a core scenario), they can click on the card and select 'Get notified about the new replies'. With more notifications types being added, we are trying to standardise a pattern that can scale and also address customer scenarios. 

Having said that, we don't want to leave you at a place where your requirements are not met. One option, I can think of right now is to give an option to broadcast the comment messages as well  to the channel, which basically means, the noise is not reduced. We will evaluate this feedback and take a decision. 

Thanks 🙏

**Comment by gecko655** [comment]
The broadcast feature is what I wanted.
I hope it will be available soon.

Thanks!

**Comment by MH4GF** [comment]
> Any participant of the thread will be automatically notified in slack.

I want to disable any threading feature too, I will describe my use case for reference. I used the posts as valuable information to keep abreast of discussions that did not directly involve me.

**Comment by shige** [comment]
> This feature is described here: https://github.com/integrations/slack#threading

I'm very confused with this release. I lost my productivity by not being able to read the comments I needed and getting more notifications I didn't need. :(

I want to manage the settings so that I can enable/disable both of the following features.

- broadcast feature
- threading feature

Thanks!

**Comment by ChrisAshworth** [comment]
Hello GitHub integration team, thanks for your work. Adding a few thoughts here.

I'm confused by this description: 

> Yes, we have recently enabled threading feature for issues. The same will be applied to PRs as well. The main reason for moving the notifications of issue into thread is to reduce the noise in the channel.

The noise in our channels has shot up since these changes were made; it is much more noisy now and hard to follow actual human conversations. Adding a thread that then posts back to the channel itself seems doubly noisy for no reason. 

Every channel we have that has a github integration is now harder to use, and we are starting to consider turning off the integration.

A related issue that came with this change is the addition of the (Comment) (Edit) (Reopen) buttons that show up in the threads, which particularly on mobile devices makes it feel like we can mess up an issue with an accidental touch. Several people on our team have reported feeling that this new addition is stressful.

Thanks for your consideration and your work.

**Comment by codeknitter** [comment]
>The noise in our channels has shot up since these changes were made; it is much more noisy now and hard to follow actual human conversations. 

Yes! I'm experiencing the same issue. 

Here is more info about our use case and set up. I hope it helps you find ways to adjust this feature to make it work better for more of us. 
- As a member of the product team I create a TON of issues and make a zillion comments during UAT, etc. 
- I must also keep track of many product-related conversations across multiple Slack channels and with many people in Slack.
- In Slack we have a channel for GitHub issues (creation and closing thereof) and a separate channel for comments only.
- I have Slack set up with @mention references so I will see in Slack when someone mentions me in a GitHub comment. Other members of the product team and QA also have this set up. 
- Previous to today:
  - I was able to easily scan the GitHub comments slack channel for comments I need to pay attention to.
  - When there were new GitHub comments, only the GitHub comments Slack channel showed new messages. I could easily ignore this one channel until I was ready to scan through all the recent comments.
  - The unread "Threads" channel just showed people-conversations, no bots or GitHub comments. This made it super easy to follow up on product discussions with coworkers. And the conversations were easy to find in Slack.
- Today:
   - In our GitHub comments Slack channel:
      - Every GitHub comment made triggers a notification saying a NEW issue was created. 
      - I'm no longer able to easily scan the channel for relevant comments since all the comments are now hiding in threads. 
      - Every time someone comments on a ticket I created I get another @mention in Slack, even if the comment is not directed at me.
      - Every comment I make in GitHub is @mentioning me in Slack. I feel like I'm talking to myself, noisily. 
   - The unread Threads channel in Slack is so flooded with GitHub @mentions that I no longer have any idea if a real person needs my attention or not. It's no longer easy to find the people-conversations that I need to follow up on in Slack.

I really appreciate all the hard work that goes into making GitHub work seamlessly with other products such as Slack. Most of the time these improvements help us be more efficient for which I am grateful! 

And I totally understand how tricky it can be to balance various use cases and business needs when building and improving features. 

I hope that I've explained our situation well enough that you'll be able to provide some way for us to adjust the GitHub-Slack integration to better fit our needs. Thank you for your time and consideration.

**Comment by tchamblee** [comment]
> The noise in our channels has shot up since these changes were made; it is much more noisy now and hard to follow actual human conversations.

🖐️ Our Slack channels too. I've been getting hammered with Slack @mentions. Half the time the mentions are as a result of my own comments in Github. 🤦

**Comment by sakimyto** [comment]
I would like this feature very much. There are many similar opinions in the Japanese engineer community...

**Comment by nielsiano** [comment]
This is an absolutely horrible change. Please at least make it an option to have all comments in the channel as before.

**Comment by EdwardSalkeld** [comment]
This really doesn't seem to be very well thought through. I don't hate the threading element, but there are a couple of absolute dealbreakers here. @codeknitter above highlighted them pretty well so I'll just stress the ones that I think are particularly mad/bad.

> Every GitHub comment made triggers a notification saying a NEW issue was created.
This is just silly. I can see the problem you're trying to solve but... they're not new issues.

> When there were new GitHub comments, only the GitHub comments Slack channel showed new messages. I could easily ignore this one channel until I was ready to scan through all the recent comments.
And this goes double for muting the notifications. I've got the GIthub App channel muted and can just check it at my leisure. Now it's shouting at me all the time.

**Comment by mgreystone** [comment]
I can't describe this frustration any better than @codeknitter either. 

However, i'm afraid i also need to double-down on how disruptive these thread notifications are. I am now being constantly pinged after business hours & had to change my DND schedule to compensate. I'm now worried i am going to miss actual messages from coworkers.

**Comment by ashokirla** [comment]
Hi all 👋 , 

As mentioned previously, we will provide an option to enable broadcasting of the comment notifications to the channel. We will rollout this support in about 2-3 weeks. With this feature, you can choose any subscription in the channel where comment notifications is enabled and decide if you also want to see the comment notifications from the thread in the direct channel stream as well. 

Thanks 🙏

**Comment by sullman** [comment]
I'm just adding noise by saying "us too!", but I feel strongly enough that maybe more noise will help convey the magnitude of the feedback here. The commitment to an option to enable broadcasting is better than the status quo, but not as helpful as an option to turn off threading completely.

@MH4GF described it very succinctly:

> I used the posts as valuable information to keep abreast of discussions that did not directly involve me.

For GHIs that I opened, am assigned, mentioned in, or otherwise subscribed to, I already have the first class notifications view in GitHub. Our slack integration was noisy on purpose because it allowed an easy way to keep abreast of the things that I _wasn't_ already subscribed to. I might see a comment on a GHI and realize that I need to be involved and want to either reply or at least subscribe to the issue. The option to broadcast all threaded replies at least helps with this (even if clumsily).

@codeknitter described the problems with threads well, so I'll just highlight the key comment to me:

> The unread "Threads" channel just showed people-conversations, no bots or GitHub comments. This made it super easy to follow up on product discussions with coworkers. And the conversations were easy to find in Slack.

This change has made my slack Threads noisier and less useful. If someone replied to a GHI I opened, I was going to see it anyway. But now it's polluting my slack Threads. And since Slack threads don't collapse the first message in a thread and new GHI messages tend to be long, the messages filling my threads are huge. These are the issues that won't be helped by simply allowing an option to broadcast all threaded replies.

**Comment by ashokirla** [comment]
@sullman We have also made a change to stop you from mentioned when you create or comment an issue. Please refer to my comment [here](https://github.com/integrations/slack/issues/1501#issuecomment-1283962776).

**Comment by gecko655** [comment]
@ashokirla 

I'm glad hearing options will be available in a couple of weeks.
But I just want to make sure one thing.
Are you planning to revert the threading feature until the broadcasting option to be available?

After I worked an entire day, I don't think we can stand the situation for weeks.
We are finally posting their GitHub comments to the Slack main timeline MANUALLY, to share their comments to the team. Our productivity has been ruined widely by this manual sharing.
I am looking for another automatic posting Slack bot to endure this painful weeks. (@everyone Let me know if you have suggestions)

**Comment by archon810** [comment]
We subscribe to the organization, not individual repos (not sure many people are aware that this is even possible).

I hope it will be possible to change the upcoming threading settings just for the organization and have it apply to all the repos.

**Comment by AshleyPinner** [comment]
There's a side effect of this that still is happening: If an issue that is assigned to someone in slack is closed, the GitHub slack integration then @'s them. For an issue that is closed. This never used to happen with the previous behaviour and is now causing excessive notifications.

Granted, development can't turn on a dime, but asking for us to put up with this for 2-3 weeks is going to get old very very very quickly.

**Comment by trevor-scheer** [comment]
We're in the process of closing out a bunch of old PRs and issues (like multiple years old). Every time one is closed, I get a message that an issue was _created_ (presumably because GH app needs a place to thread the closed message). This is really confusing and I now have a swath of "new issues" that I need to go and check that are actually very stale and recently closed.

**Comment by eloquence** [comment]
I don't think this feature is sufficiently "baked" to be the default. I would kindly request a revert rather than a preference down the road.

As others have noted, this is especially confusing during work such as backlog grooming session, where the whole point of the exercise is often to close out old issues. Slack will now suddenly display a bunch of old issues - sometimes created by folks no longer involved with a project - as "newly created". This is very confusing and disorienting, and that's not just a matter of "who moved my cheese / getting used to it" -- it's a fundamental UX problem with the way this feature is designed right now.

For this to be the default, the creation of new threads needs to IMO be more carefully thought through.

We've had folks on our team switch from Slack integration to email to get usable notifications again, and this is after days of everyone giving the feature the benefit of the doubt.

**Comment by psifertex** [comment]
Unfortunately other than the proposed fix of highlighting (which is good!) I haven't seen any indications from GH that their future plans will actually address the majority of the issues raised so I'm chiming in as well to request a different solution (preferably a quick revert and re-evaluation and better testing with other teams). 

I can live with the stressful, space-wasting buttons. They're not great but sure. 

The absolute confusion being caused by messages showing up that have nothing to do with what actually just happened is an absolute deal breaker. It's enough to want to turn off the integration and subscribe to RSS or some other work-around. It's actively harming our ability to get work done and causing misunderstandings about even little items such as which of our team members are active (a teammate on vacation popped up due to this and multiple people thought he was back and active when of course it was just this new, broken behavior).

**Comment by koic** [comment]
First of all, I would like to thank the people involved in Slack for their efforts.

There's been some discussion about it already,I expect issue comment notifications to **flow** from the Slack-GitHub integration. Not **stock** to thread of Slack.
After this change, I often miss notifications for GitHub issue comments. I think the most important thing in Slack is news flows and I can find non-news issue comments on GitHub. In other words, the roles expected of both tools are different.

Currently, a low-hanging fruit notification tool from GitHub is email. I would be happy to at least opt out of the new integration behavior. Thank you.

**Comment by pombredanne** [comment]
The new use of threads for issue comments makes the whole Slack integration utterly useless forcing context switches and providing context- and content- free channel notifications when there is a new comment issue. Using threads should be an opt-in not something you should be forced to use as Slack threads have their share (a big share) of issues out of the box.

**Comment by klovskim** [comment]
I am assigned on over 2000 issues within my organization, my slack notifications have become absolutely overwhelming. The only solution I have found is to remove myself from all issues as an assignee, which is less than ideal. Before I did this I was being pinged by the github integration roughly every 15 minutes. Not to pile on, but this is absolutely productivity-tanking.

**Comment by ashokirla** [comment]
Hi 👋 ,

We apologise for the inconvenience. We are working at a brisk pace to provide the support to publish the comment notifications to channel. Hoping to extend this feature by the end of this week. 

Thanks 🙏

**Comment by ChrisAshworth** [comment]
Hi @ashokirla,

Since there has been much discussion on this issue I wanted to highlight a detail that may have been lost, which is that at least in our case, posting comments to the channel is not a resolution to what is causing our problems. 

One of the (big) sources of noise we are encountering is the fact that the integration must generate a new thread to post _to_, which in the case of closing old issues means that closing any old issue results in:

- the message about closing the issue, but also
- a much larger message created (to create a thread) listing the entire issue that is being closed, 
- which is additionally marked as being "new", even though it is old and also instantly closed

This is why, of our organization, we would prefer to fully turn off the threading feature entirely; it worked well for us before, and it does not work well for us now, in a way that is unrelated to whether comments are posted to the channel. (We have not been using the feature to post comments to the channel, so bringing that ability back does not affect the usability of the new integration for us.)

P.S.: Also, these (comment)(edit)(close) buttons, ack!  The only thing they do for us is give us big bright noise we need to ignore and hopefully not press!

**Comment by pombredanne** [comment]
@ChrisAshworth you wrote:

> we would prefer to fully turn off the threading feature entirely; 

I second that. 

and re:
> Also, these (comment)(edit)(close) buttons, ack! The only thing they do for us is give us big bright noise we need to ignore and hopefully not press!

Ditto. This is also more than annoying. This also using a lot of vertical space that further pollutes a channel. This should be optional.

@ashokirla Can I suggest that you outline and document exactly what is the direction you are taking? It feels like you will not resolve much of the concerns expressed so far. It would be good to know ASAP. 

One alternative would be to go back to using webhooks instead for a quick return to sanity? At the moment the combo of Slack + this Github app is a net negative and a serious productivity drag.

**Comment by pombredanne** [comment]
I can confirm that dropping using this app entirely and using a webhook instead is a workaround that brings back proper control. If you have many repos and channels that's going to be quite a bit of work to setup of course.

**Comment by dylanparry** [comment]
> I can confirm that dropping using this app entirely and using a webhook instead is a workaround that brings back proper control. If you have many repos and channels that's going to be quite a bit of work to setup of course.

A bit off-topic, but do you have a link to somewhere that explains how to do this?

**Comment by pombredanne** [comment]
@dylanparry a workaround to a regression would not be off topic in my book:
1. on the slack side, go to "manage apps"
2. select custom integration. Ignore this mumbo jumbo: "We recommend replacing your custom integrations with Slack apps, which have more features and use the latest APIs. Learn more about switching to Slack apps.". duh.
3. select "Incoming WebHooks", then "add to slack"
4. In "choose a channel" enter the channel you want to have notifications posted, then "Add Incoming WebHooks integration"
5. Copy the webook URL ... e.g., "https://hooks.slack.com/services/T03Hxxxxxxxxxxxxxxxxxxx"
6. On the Github side, go to a project "Settings"
7. Select "Webhooks" then "Add webhook"
8. In the "Payload URL" field paste the webhook URL from 5. 
9.  Select "Content Type"  as application/json
10. Select which GH events you want to broadcast (defaults is just push)
11. Click "Add webhook" 

Done, now you can get back to work!

**Comment by pombredanne** [comment]
@dylanparry FWIW, it used to be the way before this app existed. This app was a nice improvement over this for a while before it started to regress.

**Comment by dylanparry** [comment]
@pombredanne cheers. Yes, I totally agree, it's become useless for productivity now.

**Comment by marcpalmer** [comment]
I too am unhappy with this change. 

This is because it totally breaks the usefulness of Slacks "Threads" pseudo-channel where I regularly return to see what others have said on threads that I am involved with.

This is now constantly spammed by Github issues and it has become impossible to find conversations that actually involve me / that I want to return to.

**Comment by dylanparry** [comment]
> 

I can't get the instructions you posted to work. I just get an error 400 in GitHub's list of recent deliveries for the webhook I've set up :confused:

**Comment by joepestro** [comment]
@pombredanne @dylanparry looks like slack's incoming webhook is looking for `text` in the JSON payload, which isn't something that's sent by this github webhook.

**Comment by ashokirla** [comment]
Hi all 👋 , 

We have now added support to also broadcast the comment notifications for issues to channel if you choose to. This support is now live.

If you have subscribed for comment notifications, and you also want to the channel members to see the comments instead of just those who are participants of the issue, you can opt-in for the same by running 
`/github subscribe org/repo comments:"channel"`

This will give you the exact experience you previously had and in addition you will get more context if you look into the thread.

Note: By default comment will only flow into the thread. And you need to explicitly run the above command to ensure the comments also start flowing into channel as well.

We apologise for any inconvenience caused with this change. We are trying to improve the customer experience by constantly innovating ourselves. All these years GitHub app is just a post master which brings you information to Slack. We believe threading and inplace create/comment/close functionality is going to enhance our GitHub app to be a multiplayer and a true bidirectional app.  We understand that it is difficult for you to adopt to the new way for collaboration. This comments:channel feature is going to help you choose to stay where you are and or tryout the new way. 

Thanks :pray

**Comment by psifertex** [comment]
"Exact same experience". So this means we won't be spammed by older messages being recreated? That doesn't sound like it's the case based on the description but maybe I misunderstand.

To be clear: if a new comment happens and I am subscribed with this new setting, will I also see an extra message making it look like the issue was created when it wasn't in fact just created?

**Comment by ashokirla** [comment]
> To be clear: if a new comment happens and I am subscribed with this new setting, will I also see an extra message making it look like the issue was created when it wasn't in fact just created?

This happens only for initial cases because we but most of meta-data only in the parent card so that your other notifications remain simple. And any notification comment/close needs a parent card. In your case, since we dont have a parent card, we are constructing that for you. Unfortunately, slack also considers as a new notification. However, if you see the title, it doesn't say 'new issue created', instead it says 'issue created'. Please bear with us on this. This only happens once for issue that is already created. Later on and for newly created issues, you will not face this problem.

**Comment by dylanparry** [comment]
It's better, but it's still not great. We need a way to disable threads altogether. I don't want to be pinged by Slack every time someone comments on an issue that I've been involved in.

**Comment by ashokirla** [comment]
Regarding mentions, this is not a new feature. We are @ mentioning you in the same places where you were previously mentioned. 

If you are a participant in an issue, by going to threads section, you will get the full picture and you can directly take action from there. This is a very powerful feature which ensures, you dont miss out on any issues that need your attention. If you have issues where you are mentioned but you dont want to participate and think it is a noise, I question the scenario in the first place. 

Having said that, if you absolutely believe that you dont need to see issues/PR updates in threads and think it is a noise, 
I can suggest a quick workaround that will ensure you get an experience exactly what you had previously (i.e. no more pings). 

Our GitHub app only [mentions](https://github.com/integrations/slack/blob/master/README.md#how-does-mentions-work) you in the Slack workspace where you logged into GitHub the latest. You can go to a least used slack workspace or personal slack workspace and log into GitHub with our GitHub app from there. Then you will not be pinged or see updates in threads in the other main workspace.

Meanwhile we will monitor the feedback and see if we should give an option to turn off mentions.

Thanks 🙏

**Comment by Takazudo** [comment]
Thank you for the great jobs!
Let me tell you how I feel. I hope you can take this as an opinion.

As for me, I don't use slack threading.
I use threading only when the two topics of talks are occured at once.
Because it's too complicated.
I understand that threading is a nice feature.
But not for me.

I just want to handle slack channels as a single thread feed.
I switch the channels to check people's comment, issue updates.
Threading brings one layer deeper architecture to the world.

I know many people use the threading of course.
And I know this is not the place about discussion about slack's threading feature.

I feel I need a option that we can turn off the threading feature.
I think there are many people like me.

**Comment by kannokanno** [comment]
Thanks for this update.
It's gotten a lot better, but I still don't feel like I'm good enough.
It would be nice to be able to turn threads off completely

I will present one use case.
Before this thread support, we sometimes created our own threads against notifications.

It's noisy if you commenting on issues, but if you want to communicate a little.
For example: "I'd like to have a web conference about this matter, is this the right time?"

This new feature makes it difficult to convey such trifles. This is because it pollutes the thread, and after commenting on the thread it is pinged.

As an aside, the size of your team will determine whether or not threads work well in the first place. I don't think threads are always effective.

Thanks.

**Comment by pombredanne** [comment]
> I can't get the instructions you posted to work. I just get an error 400 (missing text) in GitHub's list of recent deliveries for the webhook I've set up confused

@dylanparry these are working for me alright. I am getting all notifications delivered correctly with webhook :|

**Comment by combinatorial** [comment]
While the latest changes surface comments it continues to break the threads feature of Slack for how I use it. I used to use Threads to track important ongoing conversations with co-workers. Github now dominates Threads and I can no longer find the conversations that matter. To find github issues that matter I use the github website and filter issues by milestone/project label etc. We feed comments to slack to enable quick scanning and getting an overview of what's going on. The threading of an issue in Slack has no value to our team.

**Comment by pombredanne** [comment]
In https://github.com/integrations/slack/issues/1500#issuecomment-1293622938 @ashokirla scripsit:

> Regarding mentions, this is not a new feature. We are @ mentioning you in the same places where you were previously mentioned.
> 
> If you are a participant in an issue, by going to threads section, you will get the full picture and you can directly take action from there. This is a very powerful feature which ensures, you dont miss out on any issues that need your attention. If you have issues where you are mentioned but you dont want to participate and think it is a noise, I question the scenario in the first place.

This is making assumptions about a certain behaviour and workflow and that organization of this integration are looking for more than a notification of events. This does not seem to match well the reality of the  users reporting their problems here. Hijacking slack threads happens to be intrusive and spam-like noise in most cases.

> Having said that, if you absolutely believe that you dont need to see issues/PR updates in threads and think it is a noise, I can suggest a quick workaround that will ensure you get an experience exactly what you had previously (i.e. no more pings).
> 
> Our GitHub app only [mentions](https://github.com/integrations/slack/blob/master/README.md#how-does-mentions-work) you in the Slack workspace where you logged into GitHub the latest. You can go to a least used slack workspace or personal slack workspace and log into GitHub with our GitHub app from there. Then you will not be pinged or see updates in threads in the other main workspace.

The workaround  suggested looks an extremely complicated thing. This is making assumptions about the availability of multiple slack workspaces and various level of login which may or may not exist in practice. I cannot fathom how I could even start to suggest seriously to every user in a company to attempt doing such a thing in practice.

The point is that the introduced threading feature generates spam-like noise and destroys the usability of this tool, wasting a lot of time for many people. And we are several here that care enough to spend some of our time to help here and provide feedback.

Could it be possible to publish the source code of this integration? We could help fix the issues.

**Comment by codeknitter** [comment]
> I feel I need a option that we can turn off the threading feature.

I agree with @Takazudo  The threading is overloading the Slack "Threads" channel. I can no longer easily find the people-conversations I need to follow up on.

And this assumption doesn't match my needs at all:
>If you are a participant in an issue, by going to threads section, you will get the full picture and you can directly take action from there. 

I already know the "full picture" outside of Slack as these are often github issues I created and are involved in closely. I only need to know the new comment by itself. The way it was before where it just mentioned the issue name was all I needed to know to have the full context.

As for the workaround:
>Our GitHub app only [mentions](https://github.com/integrations/slack/blob/master/README.md#how-does-mentions-work) you in the Slack workspace where you logged into GitHub the latest. You can go to a least used slack workspace or personal slack workspace and log into GitHub with our GitHub app from there. Then you will not be pinged or see updates in threads in the other main workspace.

This unfortunately won't work for me. Our Slack and GitHub integration is managed at the company level with notifications for an entire repo being dropped into specific Slack channels. The only other thing I can think of to try to cut down the noise is to leave the github integrated slack channels completely and go back to relying on email notifications. Unfortunately this solution means I am likely to miss comments I need to know about on tickets that were not written by me. 

Again, I appreciate that y'all are spending time improving integrations. I just wish more diverse use-cases were included in the initial planning of this update so this would not have become an issue. 

Please let us opt out of threading. 🙏

**Comment by shige** [comment]
@ashokirla 

> > I feel I need a option that we can turn off the threading feature.
> 
> I agree with @Takazudo The threading is overloading the Slack "Threads" channel. I can no longer easily find the people-conversations I need to follow up on.

> Please let us opt out of threading. 🙏

I agree with them.

I just want it back to how it was before... 🙏

**Comment by DennisClark** [comment]
The GitHub/Slack integration is supposed to help TEAMS work together.  Teams, Teams, Teams.  Somehow that is lost with the current functionality.  Extremely frustrating.  Dreadful.  It stinis.

**Comment by bribroder** [comment]
How about diversifying this new tweak? Instead of just:

comments:"channel"

To enable comments in channel, how about controlling this behavior with comments:"threads", comments:"channel",

Allow folks to opt into just one or the other or to receive both if neither is specified

**Comment by pombredanne** [comment]
What's really painful with this comment and threading issue is that it never stops on giving. Every new day sees its new batch of confusing comments and threads spam reminding me continuously about this problem and wasting each time a bit more precious time.

@ashokirla gentle ping. Any update? Or should we really go out of our way to create an alternative integration?

**Comment by psifertex** [comment]
Having lived with this now for a while and looking through how it's impacting our organization I'd like to propose a solution. 

First, I see the value in the threading _for issues that exist in slack already_. It works well in that situation and it's why I'm sure the GH devs have been... slow to heed all of the complaints in what is clearly the most contentions issue this repo has ever seen (the only other issues with more upvotes have been around for four years and this has cracked the top six in less than two weeks). 

The main problem most folks seem to be running into is the necro-posting that is creating old issues as if they were new. Is it possible for the integration to simply... not do that? If the original issue doesn't exist, don't thread. If the original issue does, knock yourself out with the threading. That would offer a much more graceful transition to this new mechanism without a huge amount of confusion and headache.

I realize that some of the other comments above are concerned about the threading itself for other reasons but at least for some of us it's the fake "new issue" messages for old issues that is the biggest headache and source of confusion.

**Comment by eloquence** [comment]
> To enable comments in channel, how about controlling this behavior with comments:"threads", comments:"channel",

Yes exactly. Just making the behavior fully controllable would I think address everyone's concerns here: either create threads, or don't create them, based on the value of this setting. As it is, with the `comments:channel` setting I get results like this:

![Screenshot from 2022-11-01 13-47-15](https://user-images.githubusercontent.com/213636/199338223-aa597259-fc3c-44a2-b109-7dc1220d184e.png)

I'm sorry, but that doesn't help at all.  Slack still creates confusing new threads for old issues (the main problem), and now it  additionally broadcasts the comments to the channel (so we get the visual clutter of threading with the noisiness of comments - the worst of both worlds).

I understand that adding preferences like this creates maintenance debt. The alternative, I would reiterate, is to revert to the old behavior. As is, notifications are _still_ much degraded in usefulness.

@ashokirla: Further updates  on plans for this behavior would be much appreciated so we can make an informed decision on whether we need to switch to an alternative, unofficial integration.

**Comment by pombredanne** [comment]
> @ashokirla: Further updates on plans for this behavior would be much appreciated so we can make an informed decision on whether we need to switch to an alternative, unofficial integration.

gentle ping.

**Comment by vesche** [comment]
Voicing support for this issue. I also do not like this change and it's negatively impacting my workflow. It would be great if this setting was disabled by default and could be selectively toggled by workspaces who desire it.

**Comment by pombredanne** [comment]
@mgreystone FWIW, I reached out to Slack support and while they empathized, that's not their code so there is not mucho they can do.
However, they have forwarded the matter to their integrations product management team since this is not their own code.

**Comment by thijsc** [comment]
Same for us. Multiple people in our team have indicated that the current way this works is much too noisy.

**Comment by JoeFowlerJDi** [comment]
I suppose I might as well add our voice to the chorus. Everybody at our organisation hates this feature, and we are probably going to actively start looking for an alternative if there's no obvious movement on this.

In addition to the issues mentioned here already: when someone adds a comment to an old issue, where a thread does not yet exist, the message and notification makes it look like a brand new issue has been created. Very frustrating.

**Comment by specky-rum** [comment]
Another voice of dissent...

We don't use threads in Slack because as a small team we don't really need / want them. They add a level of complexity which I appreciate could be useful for some but not for us. We actively want a single stream of updates where each comment on an issue is a single update in that single, chronological thread. We have separate channels to break up the "noise" and this serves our needs perfectly. This new feature forces us to use threads which in turn makes it harder to keep up to speed with what's going on with our GitHub issues because new comments are buried away in threads that we don't look at.

We would welcome a revert of this feature or the option to switch it off so the system behaves as it did before. For us, it was much simpler and suited our needs much better than the way it works now.

**Comment by ChrisAshworth** [comment]
FWIW we've now turned off the issues notifications portion of this integration; the amount of noise introduced by these various changes has been too much for us.

**Comment by Takazudo** [comment]
@ashokirla Did you get enough feedbacks?

**Comment by pombredanne** [comment]
@ashokirla I hope you are doing OK since you seem to have been AWOL for the last couple weeks. It would be really nice to get some minimal reply, or to make this app open source so we can handle this ourselves.

In the meantime, I will try to write a script to switch to webhooks all of our repos and remove this integration entirely, but I would have mucho preferred not to have resort to this end: I started https://github.com/pombredanne/github-slack-app ... If someone here wants to band with me on this please start by adding issues with your requirements :)

**Comment by gglas-s** [comment]
So after using `/github subscribe org/repo comments:"channel"`, I now get both the comment posted to the channel (which I want), but also posts to the thread (which I don't want.)

Please allow us to disable the post to the thread in slack.

If I want to view the "thread", I will simply open the github issue by clicking the github comment link in slack, and view all the comments on the website.

**Comment by pombredanne** [comment]
FWIW, this is the last commit of this integration that was open source https://github.com/integrations/slack/tree/be7487b13833ac3cafd7ef7826fd36a87ed73701

**Comment by jplehmann** [comment]
Agreeing with OP and piling on here.

I like the Slack Github channel -- which I have muted.

However those comments showing up in Threads makes it very difficult for me to use Threads to track in-Slack convos.

We really need an option to not send GH stuff into Threads -- especially if that channel is muted!

Thanks.

**Comment by pombredanne** [comment]
I am somewhat sorry that this issue has seen no reply from GitHub for about a month. It makes me wonder. Ignoring an issue silently does not make it go away.

**Comment by pombredanne** [comment]
In trying to find a solution to this mess, I have played with the old codebase of this repo when this was still open source. This is not worthy of being furthered.

I am now investigating another avenue that would not require a separate infrastructure and would rely on GH actions only:
- https://github.com/marketplace/actions/slack-send and https://github.com/slackapi/slack-github-action which is an officially Slack-supported GH action by @stevengill @seratch @filmaj and etam 

- and the fairly well customizable trigger capabilities of GH actions documented at https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows

Let's see if I can pull a simple setup that I can document back here.

**Comment by ashokirla** [comment]
Hi 👋 , 

Thanks everyone for sharing your feedback. We have been trying to figure out how to unblock customers who are using GitHub app for Channel notifications only. And also we want to provide you a way to adopt to more collaborative threading experience whenever you are comfortable.

Based on the popular request, we are proving a **self-service way to turn off threading** in the channels where you feel it more of a noise. We have been working on this feature for the past few weeks. We will ship this feature by end of this month (Nov-30). 

Having said that, we believe threading and in-place commenting experience is the way forward for GitHub app experience in Slack and this is the recommendation by Slack team. We will share more details about the feature very soon. 

Thanks 🙏

**Comment by pombredanne** [comment]
@ashokirla Thank you. I really look forward to tomorrow's update!

You wrote:

> Having said that, we believe threading and in-place commenting experience is the way forward for GitHub app experience in Slack and this is the recommendation by Slack team. We will share more details about the feature very soon.

That's an opinion which is obviously not shared here: I can respect your belief but they do not seem to be grounded in and based on the actual feedback of you users.  Considering threading and in place commenting as the way forward is your call but it feels like a wrong direction to me, whether recommended by Slack or not. 

Could you please state clearly ASAP where you are going so we can adjust accordingly and ahead of time rather that being faced with a fait accompli? This would be much welcomed.

**Comment by JoeFowlerJDi** [comment]
> > Having said that, we believe threading and in-place commenting experience is the way forward for GitHub app experience in Slack and this is the recommendation by Slack team. We will share more details about the feature very soon.
> 
> That's an opinion which is obviously not shared here: I can respect your belief but they do not seem to be grounded in and based on the actual feedback of you users. Considering threading and in place commenting as the way forward is your call but it feels like a wrong direction to me, whether recommended by Slack or not.

This is probably drifting outside of feedback and in to general discussion, but it's no surprise that slack wants users to do everything on their system, integrate it in to their workflow and thus become more reliant on their software. I can't speak for everyone, but I am not a fan.

> 
> Could you please state clearly ASAP where you are going so we can adjust accordingly and ahead of time rather that being faced with a fait accompli? This would be much welcomed.

I'd like to second this. I'm a bit worried based on what you have said that the option to switch off threads will be obscure and therefore won't get much use, which will then be used as justification to can it down the line. If that's on the cards then fine, but I'd rather know now than have this issue pop up again later.

**Comment by electroly** [comment]
Hi @ashokirla, has the self-service option to turn off threading been delivered?

**Comment by TheMadBug** [comment]
> That's an opinion which is obviously not shared here: I can respect your belief but they do not seem to be grounded in and based on the actual feedback of you users. Considering threading and in place commenting as the way forward is your call but it feels like a wrong direction to me, whether recommended by Slack or not.

I would like to second this take given be @pombredanne. I'm aware there's a default level of resistance to any change, but this isn't some change that one or two months on we find is actually superior. It made the Slack integration go from super useful to almost pointless on day 1, and a month and a half later it's still still that way.

Our Slack channels are now full of people manually writing "Can you check the last comment on issue xxxx", because the last comment on the issue isn't appearing in a way that's easy to see or access.

**Comment by yamada-wacul** [comment]
GitHub apps for Slack still notify us SO noisy.
**WE CANNOT USE SLACK ONLY FOR GITHUB.**

I don't deny that you guys set policies on HOW YOUR SERVICE IS USED, but I totally disagree with your move to force HOW OUR USE OF SLACK and this is a serious infringement.

Please rollback IMMEDIATELY. It is extremely annoying.
I repeat: **WE CANNOT USE SLACK ONLY FOR GITHUB.**

You may only need to pay attention to GitHub, but we don't. GitHub is a tool that supports "part of our work", and we have a lot of other work to do.
Slack is a tool we use extensively for that work, and Slack is not something you can troll for your convenience.

I repeat again: **WE CANNOT USE SLACK ONLY FOR GITHUB!!!**

**Comment by ashokirla** [comment]
Hi, 

We have now added support to turn on/off threading in your channel for your issues and pr notifications. 

You can go to the channel where you don't need threading and run the following command.
`/github settings`

You will see an option to Disable/Enable threading for that channel.

<img width="825" alt="Screenshot 2022-12-02 at 10 46 51 PM" src="https://user-images.githubusercontent.com/25578249/205349352-620d0984-1737-4bb3-b359-ff9554120f56.png">


Thanks 🙏

**Comment by ChrisAshworth** [comment]
Thank you very much!

**Comment by pombredanne** [comment]
@ashokirla re:

> You can go to the channel where you don't need threading and run the following command. /github settings

Thanks, but are you saying that an admin has to go and turn this off in every channels? I appreciate your efforts but this is not a sane solution. This should be the opposite and an opt-in feature not an opt-out require extra actions everywhere. Could you please reopen this issue as I do not consider this complete as a workable solution?

**Comment by ojintoad** [comment]
Are we confident this is working as intended?

I have a case where I am seeing this:

![image](https://user-images.githubusercontent.com/3084796/206217643-1f98e9f2-005e-4d52-b3a5-210ab2e689d4.png)

However, in the same channel I ran that, we received:

1. A notification of a PR being opened in a draft state
2. A second notification, threaded, when the PR was promoted to a ready for review state

I can open a new issue if that's preferred, but it definitely seems related here so I'm starting by mentioning it in this location.

**Comment by nikhilgarg1047** [comment]
Hi @ojintoad - Can you please open a separate issue for this bug? Also, please mention the steps to reproduce for easier troubleshooting.

**Comment by ojintoad** [comment]
@nikhilgarg1047 can do thanks

**Comment by tonglil** [comment]
Connecting my Slack account to GitHub results in me being pinged in every threaded message when there is an update on GitHub.

This fills "unread threads" in Slack with so many GitHub app the threads that it becomes useless to look at. 

Do not @me on Slack if you are going to create a thread.

**Comment by mihneachirila** [comment]
it's a lot of time now that a lot of users complain about not being able to reply to a message earlier in the channel timeline. this is a big issue, I just need to be able to reply a member that wrote something some 10-15 minutes ago. **threads should be optional**.  I don't really care about slack's strategy and vision about the future. even bloody Teams allows message reply/quote, whatever this option is called. just need to be able to reply on the channel timeline, I need others to view my late reply. I get the idea of threads and it's wonderful if you really can't live without, but this makes thinks complicated. no, notifications are not a solution, they are just some sort of a workaround and it's not ok, I don't use slack for having lots of notifications, I just need to communicate simple and easy. and although the UI is almost fantastic, the messaging part has a lot to suffer.  so please, slack management, add a simple option to quote messages so that we can reply to some dude in the team that just asked a question of general interest for the entire time so that when the answer is received, everyone will be able to see it. and not having to navigate through threads, notifications, etc, then as myself where was it that I got that answer. thanks for listening to my TEDTalk

**Comment by dilburt** [comment]
Our company is currently working on a plan to migrate from Slack to Teams due to this issue. We do software development and I cannot tell you how many problems Slack's implementation of threads causes for a development team. One member asks a question of someone else in the team regarding a design decision and that person responds thinking all members of the team will get the answer but it only goes to the person who asked. Slack needs to provide an option to disable threads based on the needs of the customers, not based on their own "vision" of the product.

**Comment by djhoese** [comment]
@dilburt Does this help: https://github.com/integrations/slack#dont-want-threading

**Comment by rolyp** [comment]
@dilburt Thank you for making my day. Threading was so poorly thought out it’s been driving me nuts, especially for comment notification.

**Comment by take** [comment]
Are there any ways to turn on threading but not get spammed in the threads section by github? 🤔
